src/broker/__tests__/addPartitionsToTxn.spec.js

```diff
@@ -97,5 +97,19 @@ describe('Broker > AddPartitionsToTxn', () => {
         'The producer attempted to use a producer id which is not currently assigned to its transactional id'
       )
     )
+
+    await expect(
+      broker.addPartitionsToTxn({
+        transactionalId,
+        producerId: 'foo',
+        producerEpoch,
+        topics: [
+          {
+            topic: topicName,
+            partitions: [1, 2],
+          },
+        ],
+      })
+    ).rejects.toEqual(new SyntaxError('Cannot convert foo to a BigInt'))
```

Comment:
It doesn't feel like this belongs here. We don't really _want_ to throw a SyntaxError if you provide a producerId that can't be converted to a BigInt. It feels a bit leaky.

I'm not sure what a better behavior would look like though. Maybe a `KafkaJSNonRetriableError` saying that `Producer id has to be numeric`